truffle/com.oracle.truffle.api.instrumentation/src/com/oracle/truffle/api/instrumentation/EventNodeFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.truffle.api.instrumentation;
+
+import com.oracle.truffle.api.CompilerDirectives.CompilationFinal;
+
+/**
+ * As opposed to an {@link EventListener event listener} an {@link EventNode event node factory}
+ * gets invoked to install a {@link EventNode node} for a source location. The factory might be
+ * invoked multiple times for a single source location but the location does never change for a
+ * particular returned event node. This allows cache compilation constant values per source
```

Comment:
This may just be a personal preference. I would much rather see commentary about the (non-semantic) relationship with particular compiler technology pushed into a separate paragraph.  It may be interesting to some readers, but for many it distracts from understanding the code.